packages/node-core/src/indexer/dictionary.service.ts

```diff
@@ -140,12 +142,38 @@ export class DictionaryService implements OnApplicationShutdown {
 
   constructor(
     readonly dictionaryEndpoint: string,
+    readonly chainId: string,
     protected readonly nodeConfig: NodeConfig,
     protected readonly metadataKeys = ['lastProcessedHeight', 'genesisHash'] // Cosmos uses chain instead of genesisHash
-  ) {
+  ) {}
+
+  async init(): Promise<void> {
+    const headers = {'Content-Type': 'application/json'};
+
+    //TODO: need to use flag here
+    //TODO: add retry logic
+
+    const res = await axios.get(`${this.dictionaryEndpoint}/metadata/${this.chainId}`, {headers});
```

Comment:
Use `AuthHttpLink` instead of `AuthLink`.
AuthHttpLink include both `HttpLink` and `AuthLink`, so that you don't need to handle the `metadata` request

Reply:
Was just doing it this way before I could restructure to use async method.